pkg/gadgets/traceloop/tracer/tracer.go

```diff
@@ -151,13 +151,29 @@ func (t *Tracer) install() error {
 		})
 	}
 
-	opts := ebpf.CollectionOptions{
-		Programs: ebpf.ProgramOptions{
-			KernelTypes: btfgen.GetBTFSpec(),
-		},
+	consts := make(map[string]interface{})
+
+	// Fill the syscall filter map with the corresponding syscall numbers.
+	syscallFiltersMapSpec := spec.Maps["syscall_filters"]
+	for _, name := range t.syscallFilters {
+		if name == "" {
+			continue
+		}
+
+		number, ok := syscalls.GetSyscallNumberByName(name)
+		if !ok {
+			return fmt.Errorf("syscall %q does not exist", name)
+		}
+
+		syscallFiltersMapSpec.Contents = append(syscallFiltersMapSpec.Contents, ebpf.MapKV{
+			Key:   uint64(number),
+			Value: true,
+		})
+
+		consts["filter_syscall"] = true
 	}
```

Comment:
```suggestion
	}
	
	consts := make(map[string]interface{})
	if len(syscallFiltersMapSpec.Contents) > 0 {
		consts["filter_syscall"] = true
	}
```

To avoid setting it for each syscall?